alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/ComponentItem.java

```diff
@@ -126,7 +141,16 @@ public ComponentItem build() throws AlertException {
 
             final String additionalDataString = ComponentKey.generateAdditionalDataString(componentAttributes);
             ComponentKey key = new ComponentKey(category, componentName, componentValue, subComponentName, subComponentValue, additionalDataString);
-            return new ComponentItem(component, subComponent, componentAttributes, category, key, operation, notificationId);
+            Comparator<ComponentItem> comparator = createComparator();
+            return new ComponentItem(component, subComponent, componentAttributes, priority, category, key, operation, notificationId, comparator);
+        }
+
+        private Comparator<ComponentItem> createComparator() {
+            return Comparator.comparing(ComponentItem::getComponent)
```

Comment:
Couldn't we compare the ComponentKey before the priority?

Reply:
that certainly makes it easier.